plugins/modules/win_gpo_link.ps1

```diff
@@ -0,0 +1,278 @@
+                    $enabled = "No"
+                    $enforced = "No"
+                } elseif (($split[$s].endswith(";2"))) {
+                    $enabled = "Yes"
+                    $enforced = "Yes"
+                } elseif (($split[$s].endswith(";3"))) {
+                    $enabled = "No"
+                    $enforced = "Yes"
+                }
+                #we create an object representing each GPOs, its links status and link order
+                $return = New-Object psobject
+                $return | Add-Member -membertype NoteProperty -Name "Target" -Value $target
+                $return | Add-Member -membertype NoteProperty -Name "GPOID" -Value $gpoguid
+                $return | Add-Member -membertype NoteProperty -Name "DisplayName" -Value $gponame
+                $return | Add-Member -membertype NoteProperty -Name "Domain" -Value $gpodomain
+                $return | Add-Member -membertype NoteProperty -Name "Enforced" -Value $enforced
+                $return | Add-Member -membertype NoteProperty -Name "Enabled" -Value $enabled
+                $return | Add-Member -membertype NoteProperty -Name "Order" -Value $order
+                $return
+            }
+        }
+    }
+}
+$spec = @{
+    options = @{
+        state       = @{ type = "str"; choices = "query", "present", "absent"; default = "present" }
+        path        = @{ type = "list"; elements = "str"; required = $true }
+        gponame     = @{ type = "str"; required = $true }
+        order       = @{ type = "int" }
+        domain      = @{ type = "str" }
+        enforced    = @{ type = "bool"; default = $false }
+        linkenabled = @{ type = "bool"; default = $true }
+    }
+    supports_check_mode = $true
+}
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
+$state = $module.Params.state
+$path = $module.Params.path
+$gponame = $module.Params.gponame
+$gpoorder = $module.Params.order
+$gpodomain = $module.Params.domain
+$gposerver = $module.Params.server
+$enforcedbool = $module.Params.enforced
+$gpolinkenabledbool = $module.Params.linkenabled
+$ErrorActionPreference = 'Stop'#Error Action
+try {
+    Import-Module GroupPolicy
+} catch {
+    $module.FailJson("win_grouppolicy requires the GroupPolicy  PS module to be installed")
+}
+try {
+    Import-Module activedirectory
+} catch {
+    $module.FailJson("win_grouppolicy requires the activedirectory  PS module to be installed")
+}
+#Converting Needed Variables from bool into Strings :)
+#Microsoft specified that as string. Possible "No","yes","Unspecified" https://docs.microsoft.com/en-us/powershell/module/grouppolicy/set-gplink
+if ($gpolinkenabledbool) {
+    $gpolinkenabled = "yes"
+} else {
+    $gpolinkenabled = "no"
+}
+if ($enforcedbool) {
+    $enforced = "yes"
+} else {
+    $enforced = "no"
+}
+#States
+if ($state -eq "query") {
+    foreach ($pa in $path) {
+        [string]$path = $pa
+        if ($path) {
+            get-gplink -path $path
```

Comment:
You aren't doing anything with this result causing it to be sent back to the controller. You should avoid that at all costs. Either `| Out-Null` the result or capture in a variable if you need it for later. Based on the context it sounds like you want to return this information as a structured object. You need to store it into a var and set `$module.Result.key = 'value'` so it's returned back to the controller.

Finally we try to avoid having modules return and set information. `state=query` should be removed from this module and the query functionality should be implemented in a `win_gpo_link_info` module instead.

Reply:
I will generate a new  module for that.